client/src/components/geneExpression/index.js

```diff
@@ -1,59 +1,80 @@
 /* rc slider https://www.npmjs.com/package/rc-slider */
 
 import React from "react";
-import _ from "lodash";
 import { connect } from "react-redux";
-import HistogramBrush from "../brushableHistogram";
-import * as globals from "../../globals";
-import AddGenes from "./addGenes";
+import { Button } from "@blueprintjs/core";
+import GeneSet from "./geneSet";
+
+import CreateGenesetDialogue from "./menus/createGenesetDialogue";
+import EditGenesetNameDialogue from "./menus/editGenesetNameDialogue";
 
 @connect((state) => {
   return {
-    userDefinedGenes: state.controls.userDefinedGenes,
     differential: state.differential,
+    genesets: state.genesets.genesets,
   };
 })
 class GeneExpression extends React.Component {
+  renderGeneSets = () => {
+    const sets = [];
+    const { genesets } = this.props;
+
+    for (const [name, genes] of genesets) {
+      sets.push(
+        <GeneSet
+          key={name}
+          setGenes={Array.from(genes.genes.keys())}
+          setName={name}
+        />
+      );
+    }
+    return sets;
+  };
+
+  renderDiffexpGeneSets = () => {
+    const { differential } = this.props;
+
+    const setGenes = [];
+
+    if (differential.diffExp) {
+      differential.diffExp.forEach((diffexpGene) => {
+        setGenes.push(diffexpGene[0]);
+      });
+    }
+
+    return differential.diffExp ? (
+      <GeneSet
+        key="Temp DiffExp Set"
+        setGenes={setGenes}
+        isDiffexp
+        setName="Temp DiffExp Set"
+      />
+    ) : null;
```

Comment:
We're essentially doing the same if statement twice in succession. Maybe exit early instead